recipes/cassandra-cpp-driver/all/patches/2.15.3/fix-cmake.patch

```diff
@@ -120,7 +118,23 @@ index 06b84b759..eae72555d 100644
  file(GLOB SOURCES *.cpp *.hpp)
  
  if(APPLE)
-@@ -126,78 +161,46 @@ configure_file(
+@@ -38,15 +73,11 @@ endif()
+ 
+ add_subdirectory(third_party/curl)
+ add_subdirectory(third_party/hdr_histogram)
+-add_subdirectory(third_party/http-parser)
+-add_subdirectory(third_party/minizip)
+ add_subdirectory(third_party/sparsehash)
+ 
+ list(APPEND INCLUDE_DIRS 
+   third_party/curl
+   third_party/hdr_histogram
+-  third_party/http-parser
+-  third_party/minizip
+   third_party/mt19937_64
+   third_party/rapidjson/rapidjson
+   third_party/sparsehash/src)
```

Comment:
Can you take a look at all the third-party libraries?
- rapidjson is also available
- sparsehash isn't (yet). Can you at least add a FIXME for all the third-party libraries you were unable to use conan packages for?

Reply:
[Done](https://github.com/conan-io/conan-center-index/pull/3110/commits/33314d4e55467ecbfb220a6036edc654bf8f07bd)

Reply:
I fixed [issue](https://github.com/conan-io/conan-center-index/pull/3110#discussion_r512710485) and rapidjson from conan started to be used, and the new issue showed up: the driver uses patched rapidjson

[1](https://github.com/datastax/cpp-driver/commit/6165cd92b026be858fb61aa7efcf48d0c213dcc3#diff-fe471fe70a307f6b6981fe91376fecc27125c015201ecdcec6690959b70ba0dfR45  )

[2](https://github.com/datastax/cpp-driver/commit/6165cd92b026be858fb61aa7efcf48d0c213dcc3#diff-246be3c94a158a17d3c4e7e1a854155c7d50089091ac5de1dda3619f947dfdc2L20  )

[3](https://github.com/datastax/cpp-driver/commit/6165cd92b026be858fb61aa7efcf48d0c213dcc3#diff-f402b6106154551920256253c6e8acb0133b7b315a32597e7242e225403d2985R33)

When I build  cassandra-cpp-driver with packaged rapidjson on Linux with clang - the build fails. Ok on Windows. Ok with gcc on Linux.
Having the above we encounter the issue of not being able to use rapidjson from conan-center-index.
The issue can be solved in two ways:
1) We forget about clang on Linux and use packaged rapidjson
2) We use patched rapidjson to be able to build on Linux with clang.

Reply:
https://github.com/datastax/cpp-driver/commit/6165cd92b026be858fb61aa7efcf48d0c213dcc3#diff-f402b6106154551920256253c6e8acb0133b7b315a32597e7242e22540 f3d2985R33
The header change in ` cpp-driver/src/third_party/rapidjson/rapidjson/document.h` looks like to disable `-Wconversion` compiler warnings.
I wouldn't worry about this.

Unless cassandra-cpp-driver uses `-Werror`. If this is the case, then I would disable the `-Werror`.

Reply:
The `-Werror` is coming from here:
https://github.com/datastax/cpp-driver/blob/3d8c5abc16bc51b49de6e2931848e4c3553480be/src/CMakeLists.txt#L72-l76

Reply:
@madebr [fixed](https://github.com/conan-io/conan-center-index/pull/3110/commits/1b19c9edc1bf41a1a930b229f53ffdf2a8ec1981)